kyo-core/shared/src/test/scala/kyo/ChannelTest.scala

```diff
@@ -298,6 +347,34 @@ class ChannelTest extends Test:
                 s <- c.size
             yield assert(r == Seq(1, 2) && s == 2)
         }
+        "should consider pending puts" in pendingUntilFixed {
```

Comment:
@johnhungerford I noticed this behavior when trying to create concurrency tests with `drainUpTo`. Do you think the method should consider pending puts? It doesn't seem very useful without that.

Reply:
It's still useful I think, but I don't see why we shouldn't drain the pending puts also. I take it regular `drain` should do so as well?

Reply:
I've created https://github.com/getkyo/kyo/issues/978 for this